### DISCUSSION TITLE
Is MS finally going to fix this issue?

### DISCUSSION THREAD

**Discussion by garyd9** [kangyu-california/PersistentWindows#111] [General]
It looks like MS is finally going to fix the primary reason that PW exists for:

https://devblogs.microsoft.com/directx/avoid-unexpected-app-rearrangement/

**Comment by kangyu-california** [comment]
The new Rapid HPD patch from MS only applies to DP monitor.
The approach that PW takes is hardware agnostic and thus more general, the main focus of PW is recovery instead of preventing windows from moving.
There are quite a few scenarios where user want windows to be moved to prior position (after multiple monitor config changes).

**Comment by JohnChristmann** [comment]
FYI this feature is built into Windows 11, so you can try it out if you install the Windows Insider builds of that.  I've been very happy with it so far.

**Reply by JohnChristmann** [reply]
Also funny thing, even Microsoft's fix can't do anything about Winamp.

**Comment by 1aTa** [comment]
I've been using PW on Windows 10 for quite a while and it's been a superb utility. I was hoping Windows 11 would fix the issues I have using an ultrawide DP primary display and 16:9 HDMI secondary. Unfortunately the rapid HPD feature has not helped, if anything it's made things even worse.

Back to PW I go and it seems to be working just fine on W11. Thank you @kangyu-california for all your hard work.

**Comment by JohnChristmann** [comment]
I've had far better results with Windows 11's implementation myself.  It's still far from perfect but windows completely failing to restore is a rare occurrence instead of happening multiple times per day with PW, particularly relating to minimized windows.

**Reply by 1aTa** [reply]
Interesting. What's your display setup out of interest?

**Reply by suskozaver** [reply]
same for me, W11 acts much better, I just have this weird problem, that on Desktop 1 (out of 4), when coming out of sleep mode (turned off screens), the taskbar is behind whatever window is opened on Screen 1.
Not all windows get restored, Razer's Synapse app for example. 
Hitting Shift+Ctrl+Win+B resets something and the issue is solved.
Setup is:
Screen 1, taskbar yes, Razer (Razer Blade 15” Base (2021) | RZ09-0369x), 1920x1080 (100%)
Screen 2, taskbar no, Samsung U32r59x, 3840x2160 (125%)

**Reply by kangyu-california** [reply]
@suskozaver I am not sure the taskbar hidden issue is with PW or with Win11. In either case, I would suggest add -fix_zorder=1 to PW command line, or remove -fix_zorder=1 if it is already specified.

**Comment by JohnChristmann** [comment]
Basically the same setup as yours, but both are 16:9.  Everything restores correctly the vast majority of the time, though some applications seem to try to correct their position on their own and screw it up occasionally.  It's very rare I have a complete failure to restore certain windows (like once a month maybe?)
